src/main/java/com/azure/cosmos/examples/bulk/async/SampleBulkQuickStartAsync.java

```diff
@@ -0,0 +1,250 @@
+                    + " in " + "database " + database.getId() + ". e: " + exception.getLocalizedMessage());
+            return Mono.empty();
+        }).block();
+
+    }
+
+    private void bulkCreateItems(Flux<Family> families) {
+        Flux<CosmosItemOperation> cosmosItemOperations =
+                families.map(family -> CosmosBulkOperations.getCreateItemOperation(family,
+                        new PartitionKey(family.getLastName())));
+        container.executeBulkOperations(cosmosItemOperations).blockLast();
+    }
+
+    private void bulkDeleteItems(Flux<Family> families) {
+        Flux<CosmosItemOperation> cosmosItemOperations =
+                families.map(family -> CosmosBulkOperations.getDeleteItemOperation(family.getId(),
+                        new PartitionKey(family.getLastName())));
+        container.executeBulkOperations(cosmosItemOperations).blockLast();
+    }
+
+    private void bulkUpsertItems(Flux<Family> families) {
+        Flux<CosmosItemOperation> cosmosItemOperations =
+                families.map(family -> CosmosBulkOperations.getUpsertItemOperation(family,
+                        new PartitionKey(family.getLastName())));
+        container.executeBulkOperations(cosmosItemOperations).blockLast();
+    }
+
+    private void bulkReplaceItems(Flux<Family> families) {
+        Flux<CosmosItemOperation> cosmosItemOperations =
+                families.map(family -> CosmosBulkOperations.getReplaceItemOperation(family.getId(), family,
+                        new PartitionKey(family.getLastName())));
+        container.executeBulkOperations(cosmosItemOperations).blockLast();
+    }
+
+    private void bulkCreateItemsWithResponseProcessing(Flux<Family> families) {
+        Flux<CosmosItemOperation> cosmosItemOperations =
+                families.map(family -> CosmosBulkOperations.getCreateItemOperation(family,
+                        new PartitionKey(family.getLastName())));
+        container.executeBulkOperations(cosmosItemOperations).flatMap(cosmosBulkOperationResponse -> {
+            CosmosBulkItemResponse cosmosBulkItemResponse = cosmosBulkOperationResponse.getResponse();
+            CosmosItemOperation cosmosItemOperation = cosmosBulkOperationResponse.getOperation();
+
+            if (cosmosBulkOperationResponse.getException() != null) {
+                logger.error("Bulk operation failed", cosmosBulkOperationResponse.getException());
+            } else if (cosmosBulkOperationResponse.getResponse() == null || !cosmosBulkOperationResponse.getResponse().isSuccessStatusCode()) {
+                logger.error("The operation for Document ID: {}  Document PK: {} did not complete successfully with " +
+                                "a" + " {} response code.", cosmosItemOperation.getId(),
+                        cosmosItemOperation.getPartitionKeyValue(), cosmosBulkItemResponse.getStatusCode());
+            } else {
+                //TODO why is the ID null?
```

Comment:
Is this TODO still relevant? We should either fix it or remove it.

Reply:
ID cannot be set on the CosmosItemOperation Object. Updated the code to retrieve the ID and PK value from the Item itself .
```
          logger.info("Document ID: {}  Document PK: {}", cosmosItemOperation.<Family>getItem().getId(),
                        cosmosItemOperation.<Family>getItem().getLastName());
```